### ISSUE TITLE
[Epic] 2.3: Basic distributed Store services

### ISSUE THREAD

**Issue by jm-clius** [waku-org/pm#64]
**Planned start date**:
**Due date**: 31 Dec 2023

# Summary

Ability to discover Store nodes that provide store services for specific content topics and time ranges and for service nodes to advertise their store services as such. This includes work to allow Store service nodes to only store messages for specific content topics/applications.

# Acceptance Criteria

# Tasks

# RAID (Risks, Assumptions, Issues and Dependencies)

**Comment by DeBelg** [comment]
https://github.com/waku-org/research/issues/14 linking this

**Comment by fryorcraken** [comment]
Not sure a due date makes sense on this epic, WDYT @jm-clius 
Tracked with roadmap https://github.com/waku-org/pm/issues/57

Work has started see https://github.com/waku-org/nwaku/issues/2112

**Comment by fryorcraken** [comment]
> Ability to discover Store nodes that provide store services for specific content topics

What about filter and light push? Shall we track work to filter content topic? Not part of Gen 0?  @jm-clius

**Comment by jm-clius** [comment]
Right, I think we should move this out of the Gen 0 milestone as it's not critical for network launch.

> What about filter and light push?

Filter and lightpush are already quite distributed as we can make the simplifying assumption that we can request these services from any discoverable relay node with filter and lightpush enabled (in other words, no extra work to be done). For store we still practically rely only on statically configured store nodes as there's no in-band way to determine if a selected Store node really has the data we're interested in available.

**Comment by fryorcraken** [comment]
Are we happy to not have "enable filter/light push only for given content topics" not on offer at the moment?

I am fine with that but we may have review if there is a demand/push for it

**Comment by jm-clius** [comment]
> I am fine with that but we may have review if there is a demand/push for it

I think it's fine to leave out and review when requested by apps.

**Comment by fryorcraken** [comment]
This epic has been removed from the "gen 0" milestone as it is not critical for launch of the gen 0.
For clarity purposes, we being stricted on the work tracked under each milestone.
